students/hatims/Session05/mailroom.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+import sys
+"""
+It should have a data structure that holds a list of your donors and a history of the amounts they have donated. This structure should be populated at first with at least five donors, with between 1 and 3 donations each.
+You can store that data structure in the global namespace.
+The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”)
+"""
+
+donors = dict()
+donors.update({'William Gates': [7897100.00, 5000.00, 12000000.00]})
+donors.update({'Mark Zuckerberg': [19000.00, 8900000.00]})
+donors.update({'Jeff Bezos': [1000.00, 1445600.00]})
+donors.update({'Paul Allen': [9000.00, 29000000.00, 5690000000.00]})
+
+
+"""
+Creating the Report
+"""
+
+"""function will print the report header and details for the donors"""
+def create_report():
+    print_header()
+    print_details()
+
+"""function will print the report header"""	
+def print_header():
+    print("{:<20}{:5}{:15}{:5}{:10}{:5}{:15}".format('Donor Name', '|', 'Total Given', '|', 'Num Gifts', '|', 'Average Gift'))
+    print("-" * 77)
+
+"""function will print the report details for each donor"""		
+def print_details():
+    total_given=0
+    num_gifts=0
+    average_gift=0
+    for donor_name in donors:
+        for val in donors[donor_name]:
+            total_given = total_given+val
+        num_gifts = len(donors[donor_name])
+        average_gift = total_given / (num_gifts)
+        print("{:<22}${:>15.2f}{:^20}${:>15.2f}".format(donor_name, total_given, num_gifts, average_gift)) 	
+        total_given = 0		
+
+"""
+Sending a Thank You
+"""
+def send_thankyou():
+    full_name = propmpt_fullname()
+    if full_name.upper() == 'LIST':
```

Comment:
Is there a typo in the spelling of prompt here, or is that also how you spelled the function?